src/Core/Managed/Shared/Channel/IDeepCloneable.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.ApplicationInsights.Channel
+{
+    /// <summary>
+    /// Indicates that an object support deep cloning.
+    /// </summary>
+    /// <typeparam name="T">The type of the object that results from the cloning operation.</typeparam>
+    public interface IDeepCloneable<out T>
+    {
+        /// <summary>
+        /// Clones the object deeply, so that the original object and its clones share no state 
+        /// and can be modified independently.
+        /// </summary>
+        /// <returns>The cloned object.</returns>
+        T DeepClone();
```

Comment:
Sergey, please confirm that 3rd party implementing a custom telemetry class is a non-scenario for AI SDK. If that is the case, adding deep cloning capability directly to ITelemetry sounds like a good idea